src/components/tool/guide/EresearchJob/EresearchInstructions.js

```diff
@@ -12,31 +12,50 @@ import React from "react";
 import InstructionInput from "../Instructions/InstructionInput";
 
 const TEXT_PATTERN = /^[a-z0-9]*$/;
+const JOB_NAME_PATTERN = /^(?=.*[a-z0-9-])[a-z0-9-]+$/;
 
 
 export function EresearchInstructions({config}) {
     const [username, setUsername] = useLocalStorageState("qutUsername", "qutusername");
+    const [batchJobName, setBatchJobName] = useLocalStorageState("qutBatchJobName", "batch-job-001");
 
     return <Box>
         {config.service === "Lyra" &&
-            <InstructionInput
-                label="QUT Username"
-                placeholder="username"
-                value={username}
-                onChange={setUsername}
-                pattern={TEXT_PATTERN}
-                helperText="Optionally enter your QUT username so that it can be pre-populated in the commands below."
-                errorText="Username must only contain lowercase letters and numbers"
+            <>
+                <InstructionInput
+                    label="QUT Username"
+                    placeholder="username"
+                    value={username}
+                    onChange={setUsername}
+                    pattern={TEXT_PATTERN}
+                    helperText="Optionally enter your QUT username so that it can be pre-populated in the commands below."
+                    errorText="Username must only contain lowercase letters and numbers"
+                />
+                {config.jobType === "Batch" &&
+                    <InstructionInput
+                        label="Batch Job Name"
+                        placeholder="batch-job-001"
+                        value={batchJobName}
+                        onChange={setBatchJobName}
+                        pattern={JOB_NAME_PATTERN}
+                        helperText="Please enter your batch job name so that it can be pre-populated in the commands below."
+                        errorText="Job name must only contain lowercase letters, numbers, '-' and must not be empty."
+                    />
+                }
+            </>
+        }
+        {batchJobName !== '' &&
```

Comment:
`batchJobName !== ''` is evaluated 3 times so can be turned into `const isBatchJob`, or if it's truthy, possibly replace the expression with `batchJobName`

Reply:
I was thinking the same when I implement this. It turns out if you turn `batchJobName !== ''` into a `const` function the function wouldn't get updated unless we utilise `useEffect` that will possibly make the code unnecessarily complicated?

Reply:
Updated and removed `!== ''`

Reply:
I was suggesting just `const isBatchJob = batchJobName !== ''`, but just relying on the truthy value is fine :+1: